### PR TITLE
Fix nil pointer dereference in cloud_cidr_allocator.

### PR DESCRIPTION
**Pull request by Dirbaio**
This PR fixes a nil pointer dereference that can happen when you run kube-controller-manager with --cidr-allocator-type=CloudAllocator and there are nodes where the cloud provider can't get the alias ranges for whatever reason.

This happened to me when switching my GCE cluster from using cloud routes to IP aliases. I added the flag but I had some GCE instances without the IP alias configured. 

**Release note**:

```release-note
Fix nil pointer dereference when running controller-manager with --cidr-allocator-type=CloudAllocator
```

### CODE REVIEW

File: pkg/controller/node/ipam/cloud_cidr_allocator.go [hunk 1]
```diff
@@ -197,11 +197,21 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 
 	cidrs, err := ca.cloud.AliasRanges(types.NodeName(nodeName))
 	if err != nil {
-		util.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
+		node, err = ca.nodeLister.Get(nodeName)
```
Comment (bowei) [inline pkg/controller/node/ipam/cloud_cidr_allocator.go]: can we combine the error cases into one block:
```
if err != nil || len(cidrs) == 0 {
  ....
}
```
Looks like the same code

Comment (Dirbaio) [reply pkg/controller/node/ipam/cloud_cidr_allocator.go]: Done! The error is different though so it requires an extra if inside.

**Comment by Dirbaio** [comment]
Rebased and squashed

**Comment by Dirbaio** [comment]
Rebased and fixed the code so it doesn't overwrite `err` with the get node error.

/assign @bowei

**Comment by bowei** [comment]
@jingax10 can you take a look?

**Comment by jingax10** [comment]
@Dirbaio: sound like the issue is already fixed by https://github.com/kubernetes/kubernetes/pull/58186. Please confirm. Thanks!

**Comment by Dirbaio** [comment]
I haven't actually tested to reproduce the error (it's quite involved) but the diff looks like it fixes the exact issue, yup.

Closing.

**Pull request closed without merging**